tree/dataframe/test/dataframe_snapshot_ntuple.cxx

```diff
@@ -0,0 +1,447 @@
+#include "ROOT/RNTupleWriter.hxx"
+#include "ROOT/RNTupleReader.hxx"
+#include "ROOT/RNTupleInspector.hxx" // For testing compression settings
+
+#include "TROOT.h"
+#include "TSystem.h"
+
+#include "gtest/gtest.h"
+#include "NTupleStruct.hxx"
+
+using ROOT::Experimental::RNTupleInspector;
+using ROOT::Experimental::RNTupleModel;
+using ROOT::Experimental::RNTupleReader;
+using ROOT::Experimental::RNTupleWriter;
+
+using namespace ROOT::RDF;
+
+class FileRAII {
+private:
+   std::string fPath;
+
+public:
+   explicit FileRAII(const std::string &path) : fPath(path) {}
+   FileRAII(FileRAII &&) = default;
+   FileRAII(const FileRAII &) = delete;
+   FileRAII &operator=(FileRAII &&) = default;
+   FileRAII &operator=(const FileRAII &) = delete;
+   ~FileRAII() { std::remove(fPath.c_str()); }
+   std::string GetPath() const { return fPath; }
+};
+
+TEST(RDFSnapshotRNTuple, FromScratchTemplated)
+{
+   FileRAII fileGuard{"RDFSnapshotRNTuple_from_scratch_templated.root"};
+   const std::vector<std::string> columns = {"x"};
+
+   auto df = ROOT::RDataFrame(25ull).Define("x", [] { return 10; });
+
+   RSnapshotOptions opts;
+   opts.fOutputFormat = ROOT::RDF::ESnapshotOutputFormat::kRNTuple;
+
+   auto sdf = df.Snapshot<int>("ntuple", fileGuard.GetPath(), columns, opts);
+
+   EXPECT_EQ(columns, sdf->GetColumnNames());
+
+   auto ntuple = RNTupleReader::Open("ntuple", fileGuard.GetPath());
+   EXPECT_EQ(25ull, ntuple->GetNEntries());
+
+   auto x = ntuple->GetView<int>("x");
+   for (const auto i : ntuple->GetEntryRange()) {
+      EXPECT_EQ(10, x(i));
+   }
+}
+
+TEST(RDFSnapshotRNTuple, FromScratchJITted)
+{
+   FileRAII fileGuard{"RDFSnapshotRNTuple_from_scratch_jitted.root"};
+   const std::vector<std::string> columns = {"x"};
+
+   auto df = ROOT::RDataFrame(25ull).Define("x", [] { return 10; });
+
+   RSnapshotOptions opts;
+   opts.fOutputFormat = ROOT::RDF::ESnapshotOutputFormat::kRNTuple;
+
+   auto sdf = df.Snapshot("ntuple", fileGuard.GetPath(), "x", opts);
+
+   EXPECT_EQ(columns, sdf->GetColumnNames());
+
+   auto ntuple = RNTupleReader::Open("ntuple", fileGuard.GetPath());
+   EXPECT_EQ(25ull, ntuple->GetNEntries());
+
+   auto x = ntuple->GetView<int>("x");
+   for (const auto i : ntuple->GetEntryRange()) {
+      EXPECT_EQ(10, x(i));
+   }
+}
+
+void BookLazySnapshot(std::string_view filename)
+{
+   auto d = ROOT::RDataFrame(1);
+   ROOT::RDF::RSnapshotOptions opts;
+   opts.fOutputFormat = ROOT::RDF::ESnapshotOutputFormat::kRNTuple;
+   opts.fLazy = true;
+   d.Snapshot<ULong64_t>("t", filename, {"rdfentry_"}, opts);
+}
+
+TEST(RDFSnapshotRNTuple, LazyNotTriggered)
+{
+   FileRAII fileGuard{"lazysnapshotnottriggered_shouldnotbecreated.root"};
+   ROOT_EXPECT_WARNING(BookLazySnapshot(fileGuard.GetPath()), "Snapshot",
+                       "A lazy Snapshot action was booked but never triggered.");
+   // This returns FALSE if the file IS there
+   EXPECT_TRUE(gSystem->AccessPathName(fileGuard.GetPath().c_str()));
```

Comment:
This might have got lost https://github.com/root-project/root/pull/15750#discussion_r1741596237